cmd/explaintest/main.go

```diff
@@ -649,26 +650,33 @@ func loadAllTests() ([]string, error) {
 // openDBWithRetry opens a database specified by its database driver name and a
 // driver-specific data source name. And it will do some retries if the connection fails.
 func openDBWithRetry(driverName, dataSourceName string) (mdb *sql.DB, err error) {
+	now := time.Now()
 	startTime := time.Now()
-	sleepTime := time.Millisecond * 500
-	retryCnt := 60
-	// The max retry interval is 30 s.
-	for i := 0; i < retryCnt; i++ {
+	sleepTime := time.Millisecond * 100
+	maxSleepTime := time.Second * 2
+	// The max retry interval is 60 s.
+	maxDuration := time.Minute
+	for {
+		if time.Since(now) > maxDuration {
+			break
+		}
 		mdb, err = sql.Open(driverName, dataSourceName)
 		if err != nil {
-			log.Warn("open DB failed", zap.Int("retry count", i), zap.Error(err))
+			log.Warn("open DB failed", zap.Duration("retry duration", time.Since(now)), zap.Error(err))
 			time.Sleep(sleepTime)
 			continue
 		}
 		err = mdb.Ping()
 		if err == nil {
 			break
 		}
-		log.Warn("ping DB failed", zap.Int("retry count", i), zap.Error(err))
+		log.Warn("ping DB failed", zap.Duration("retry duration", time.Since(now)), zap.Error(err))
 		if err1 := mdb.Close(); err1 != nil {
 			log.Error("close DB failed", zap.Error(err1))
 		}
 		time.Sleep(sleepTime)
+		sleepTime = sleepTime * 2
```

Comment:
Maybe we need put it to line660？Otherwise, we can't update the `sleepTime` value in line666.

Reply:
Done